administration/src/project-configs/bayern/config.ts

```diff
@@ -1,13 +1,40 @@
+import { GeneralJsonField, JsonField } from '../../bp-modules/applications/JsonFieldView'
 import BavariaCardTypeExtension from '../../cards/extensions/BavariaCardTypeExtension'
 import RegionExtension from '../../cards/extensions/RegionExtension'
 import { ProjectConfig } from '../getProjectConfig'
 import { DataPrivacyAdditionalBaseText, DataPrivacyBaseText, dataPrivacyBaseHeadline } from './dataPrivacyBase'
 import pdfConfiguration from './pdf'
 
+const findValue = (object: JsonField<'Array'>, key: string): GeneralJsonField | undefined => {
+  return object.value.find(entry => entry.name === key)
+}
+
+export const applicationJsonToCardQuery = (json: JsonField<'Array'>): string => {
+  const query = new URLSearchParams()
+  const personalData = findValue(json, 'personalData') as JsonField<'Array'>
+  const cardType = findValue(json, 'cardType') as JsonField<'String'>
+  const forenames = findValue(personalData, 'forenames')
+  const surname = findValue(personalData, 'surname')
+
+  if (forenames && surname) {
+    query.set(config.card.nameColumnName, forenames.value + ' ' + surname.value)
+  }
+
+  if (cardType) {
+    const cardTypeExtensionIdx = config.card.extensions.findIndex(ext => ext === BavariaCardTypeExtension)
+    const value = cardType.value === 'Goldene Ehrenamtskarte' ? 'Goldkarte' : 'Standard'
+    query.set(config.card.extensionColumnNames[cardTypeExtensionIdx] ?? '', value)
+  }
+
+  return '?' + query.toString()
+}
```

Comment:
I think it would be good, if we could handle the application json a bit more carefully: 
The original reason for the unstructured data was that we might change the application form in the future (as requirements change), but we also need to keep old applications using the old version of the application form.
Hence, there might be applications of different "format" at the same time.

Imo, it would be best if we could explicitly model the error-path here by having `applicationJsonToCardQuery` return `null` if the card query cannot be created (e.g. because some important property like name or card type cannot be determined) and then disable the button / link.

More concretely, imo, we should not use `as JsonField<'xyz'>` here. Instead, afaik, you can use `if (jsonfield.type !== 'xyz') return null`  to check if the field has the correct type and TS does its type narrowing magic.